src/Haipa.Modules.SSL/CreateCertificate.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Net;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Haipa.Modules.SSL
+{
+    public static class CreateCertificate
+    {
+        public static X509Certificate2 Create(CertificateOptions options)
+        {
+            SubjectAlternativeNameBuilder sanBuilder = new SubjectAlternativeNameBuilder();
+            sanBuilder.AddIpAddress(IPAddress.Loopback);
+            sanBuilder.AddIpAddress(IPAddress.IPv6Loopback);
+            sanBuilder.AddDnsName("localhost");
+            sanBuilder.AddDnsName(options.Issuer);
+
+            X500DistinguishedName distinguishedName = new X500DistinguishedName($"CN=" + options.Issuer);
+
+            var csp = new CspParameters
+            {
+                Flags = CspProviderFlags.UseArchivableKey | CspProviderFlags.UseMachineKeyStore
+            };
+            using (RSA rsa = new RSACryptoServiceProvider(2048, csp))
+            {
+                CertificateRequest req = new CertificateRequest(distinguishedName, rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+
+                req.CertificateExtensions.Add(sanBuilder.Build());
+
+                req.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(false, false, 0, false));
+
+                req.CertificateExtensions.Add(
+                    new X509KeyUsageExtension(X509KeyUsageFlags.KeyAgreement | X509KeyUsageFlags.DataEncipherment | X509KeyUsageFlags.KeyEncipherment, false));
+
+                req.CertificateExtensions.Add(
+                    new X509EnhancedKeyUsageExtension(new OidCollection { new Oid("1.3.6.1.5.5.7.3.1") }, true));
+
+                req.CertificateExtensions.Add(
+                    new X509SubjectKeyIdentifierExtension(req.PublicKey, false));
+
+                X509Certificate2 cert = req.Create(
+                    CreateCA(options),
```

Comment:
CA should not be recreated for every new certificate. Instead CA cert has to be reused as long it is still valid.